sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -1869,107 +1915,225 @@ class CompactBlockProcessor internal constructor(
                 emit(SyncingResult.EnhanceSuccess)
             }
 
-        private suspend fun enhanceTransaction(
-            transaction: DbTransactionOverview,
+        private suspend fun processTransparentAddressTxids(
+            transactionRequest: TransactionDataRequest.SpendsFromAddress,
             backend: TypesafeBackend,
-            downloader: CompactBlockDownloader
+            downloader: CompactBlockDownloader,
+            network: ZcashNetwork
         ): SyncingResult {
-            Twig.debug {
-                "Starting enhancing transaction (txid:${transaction.txIdString()}  block:${transaction
-                    .minedHeight})"
+            Twig.debug { "Starting to get transparent address transactions ids" }
+
+            val traceScope = TraceScope("CompactBlockProcessor.processTransparentAddressTxids")
+            val result =
+                try {
+                    // Fetching transactions is done with retries to eliminate a bad network condition
+                    val listOfResults =
+                        getTransparentAddressTransactions(
+                            transactionRequest = transactionRequest,
+                            downloader = downloader,
+                        )
+
+                    Twig.debug { "Obtained ${listOfResults.size} transactions for processing" }
+
+                    listOfResults.forEach { rawTransactionUnsafe ->
+                        // Decrypting and storing transaction is run just once, since we consider it more stable
+                        Twig.verbose { "Decrypting and storing transaction" }
+                        decryptTransaction(
+                            rawTransaction =
+                                RawTransaction.new(
+                                    rawTransactionUnsafe = rawTransactionUnsafe,
+                                    network = network
+                                ),
+                            backend = backend
+                        )
+                    }
+
+                    Twig.debug { "Done Decrypting and storing of all transaction" }
+                    SyncingResult.EnhanceSuccess
+                } catch (exception: CompactBlockProcessorException.EnhanceTransactionError) {
+                    SyncingResult.EnhanceFailed(null, exception)
+                }
+            traceScope.end()
+            return result
+        }
+
+        @Throws(EnhanceTxDownloadError::class)
+        private suspend fun getTransparentAddressTransactions(
+            transactionRequest: TransactionDataRequest.SpendsFromAddress,
+            downloader: CompactBlockDownloader,
+        ): List<RawTransactionUnsafe> {
+            val transactionsResult = arrayListOf<RawTransactionUnsafe>()
+            val traceScope = TraceScope("CompactBlockProcessor.getTransparentAddressTransactions")
+
+            if (transactionRequest.endHeight == null) {
```

Comment:
I'm not sure about this check. What should we do when a `TransactionRequest.SpendsFromAddress` with a null endHeight comes? We could use the current chain tip, the end of the current enhancing batch of blocks, or omit the request as we did now.

Reply:
I would not implement this as a function that returns a list; instead, I would implement it in a way that as each transaction is retrieved from the stream, it is sent to the back end for processing.

Reply:
Ok, I'm going to change it. And when a `TransactionRequest.SpendsFromAddress` with a null `endHeight` comes, should we omit the one particular request, or should we do any of the suggested above?

Reply:
At least for right now, the end height will always be present, so maybe this doesn't matter for right now, but what I would do if there is no end height is just process the full stream that is returned from the `GetTaddressTxids` call incrementally.

Reply:
Yes, with the commit https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/pull/1559/commits/94b6579cdf436b680671000c5cafcedd57d24a6c  we switched from list to stream obtained from `GetTaddressTxids`, but if we got `TransactionDataRequest.SpendsFromAddress.endHeight == null` from `backend.transactionDataRequests()` we can't init the stream, so we handle it like [this](https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/blob/94b6579cdf436b680671000c5cafcedd57d24a6c/sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt#L1926), which should be fine for now.